### PR TITLE
Refactor(plugins): Move jinja filter code for arista.avd.encrypt and arista.avd.decrypt to PyAVD

### PR DESCRIPTION
**Pull request by MaheshGSLAB**
## Change Summary

Move jinja filter code for arista.avd.encrypt and arista.avd.decrypt to PyAVD

## Related Issue(s)

Fixes #<ISSUE ID>

## Component(s) name

`arista.avd.<role-name>`

## Proposed changes
Moving the code and unit tests for arista.avd.encrypt and arista.avd.decrypt filter to PyAVD

## How to test
Run the molecules or CI will catch the changes

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/plugins/filter/decrypt.py [hunk 1]
```diff
@@ -1,8 +1,21 @@
 # Copyright (c) 2023-2024 Arista Networks, Inc.
 # Use of this source code is governed by the Apache License 2.0
 # that can be found in the LICENSE file.
-from ansible_collections.arista.avd.plugins.plugin_utils.errors import AristaAvdError, AristaAvdMissingVariableError
-from ansible_collections.arista.avd.plugins.plugin_utils.password_utils import METHODS_DIR
+from ansible.errors import AnsibleFilterError
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/plugins/filter/decrypt.py:4]: TODO: Need to work on fix the CI and doc string of methods

File: python-avd/pyavd/_utils/password_utils/password_utils.py [hunk 2]
```diff
@@ -11,8 +11,6 @@
 
 import base64
 
-from ansible_collections.arista.avd.plugins.plugin_utils.errors import AristaAvdError
-
 try:
```
Comment (ClausHolbechArista) [inline python-avd/pyavd/_utils/password_utils/password_utils.py]: We don't want the try/except for imports in pyavd. Also remove all the `if not HAS_CRYPTOGRAPHY` below. It is a special workaround needed for Ansible.

File: python-avd/tests/pyavd/j2filters/test_encrypt.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
```
Comment (gmuloc) [inline python-avd/tests/pyavd/j2filters/test_encrypt.py]:
```suggestion
from __future__ import annotations

```

File: python-avd/tests/pyavd/j2filters/test_decrypt.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
```
Comment (gmuloc) [inline python-avd/tests/pyavd/j2filters/test_decrypt.py]:
```suggestion
from __future__ import annotations

```

File: python-avd/tests/pyavd/utils/password/test_password.py [hunk 5]
```diff
@@ -5,14 +5,9 @@
 
 __metaclass__ = type
```
Comment (gmuloc) [inline python-avd/tests/pyavd/utils/password/test_password.py]: Please remove __metaclass__ and repalce from __future__ with `from __future__ import annotations`

File: python-avd/tests/pyavd/utils/password/test_password_utils.py [hunk 6]
```diff
@@ -6,8 +6,7 @@
 __metaclass__ = type
```
Comment (gmuloc) [inline python-avd/tests/pyavd/utils/password/test_password_utils.py]: same

**Review by ClausHolbechArista — APPROVED** [review]

**Pull request merged**